lib/pbench/agent/tool_meister.py

```diff
@@ -418,25 +496,15 @@ class PersistentTool(Tool):
 
     _tool_type = "Persistent"
 
-    def __init__(self, name, tool_opts, logger=None, **kwargs):
-        super().__init__(name, tool_opts, logger=logger, **kwargs)
+    def __init__(self, name, tool_opts, **kwargs):
+        super().__init__(name, tool_opts, **kwargs)
         self.args = None
         self.process = None
 
-    def log_subprocess_output(self, pipe: subprocess.PIPE):
-        """Thread start function to log outputs from a given pipe.
-        """
-        logger = self.logger.getChild(self.__class__.__name__)
-        for line in pipe.readlines():
-            logger.info(line)
-
     def start(self, env=None):
         assert self.args is not None, "Logic bomb!  {self.name} install had failed!"
-        assert (
-            self.tool_dir is not None
-        ), "Logic bomb!  No persistent tool directory provided!"
-        self.tool_dir = self.tool_dir / self.name
-        self.tool_dir.mkdir()
+        tool_dir = self.tool_dir / self.name
```

Comment:
The `TransientTool` asserts that `self.tool_dir` isn't `None`, but the persistent tool class just uses it.

Since both branches require it, seems this should really be checked once in the `Tool` constructor rather than deferring it until later when a problem will be harder to track down.

Reply:
Not exactly what I'd intended, but since it appears you're going to change all this anyway in the next PR... "whatever" ...